monix-eval/shared/src/main/scala/monix/eval/TaskLocal.scala

```diff
@@ -142,13 +151,13 @@ final class TaskLocal[A] private (default: => A) {
     * in the [[Task]] context.
     *
     * {{{
-    *   val local = TaskLocal(0)
-    *
     *   // Should yield 200 on execution, regardless of what value
     *   // we have in `local` at the time of evaluation
-    *   val task = local.bindL(Task.eval(100))(Task {
-    *     local.get * 2
-    *   })
+    *   val task: Task[Int] =
+    *     for {
+    *       local <- TaskLocal(0)
```

Comment:
I am worried that the constructor yields a `Task[TaskLocal[T]]`. The thing is that the `TaskLocal` constructor was used for initializing the local, the read and write operations were the ones that lifted the access of the underlying local into a `Task`.  It means that even if I want to expose the local from the creation, I would get a `Task[Task[Local]]` which will always require a flatmap operation. Why did it change?

Reply:
That's related to #539

Reply:
@Avasil I missed that one! Understood!

Reply:
@leandrob13 yeah, it's related to purity. With `Task` you can do `.memoize` btw and `flatMap` isn't that expensive.